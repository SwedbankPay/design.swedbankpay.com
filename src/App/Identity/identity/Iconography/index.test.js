import React from "react";
import { shallow } from "enzyme";

import Icons, { MaterialIcons, CardIcons, PaymentIcons, Flags } from "./index";

describe("Core: Iconography", () => {
    it("is defined", () => {
        expect(Icons).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Icons />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("MaterialIcons", () => {
        it("is defined", () => {
            expect(MaterialIcons).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<MaterialIcons />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("CardIcons", () => {
        it("is defined", () => {
            expect(CardIcons).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<CardIcons />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("PaymentIcons", () => {
        it("is defined", () => {
            expect(PaymentIcons).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<PaymentIcons />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("Flags", () => {
        it("is defined", () => {
            expect(Flags).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<Flags />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
