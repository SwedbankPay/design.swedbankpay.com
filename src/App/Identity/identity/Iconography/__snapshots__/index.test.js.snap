// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Core: Iconography CardIcons renders 1`] = `
<section>
  <h2
    id="card-icons"
  >
    Card icons
  </h2>
  <p>
    Card icons are typically used to show users which card providers that are available. The payment-card icon can be used as a placeholder and/or when the card is not recognized. 
  </p>
  <h3>
    Example of how to implement card icons
  </h3>
  <p>
    To use an icon, provide the following markup: 
    <CodeTags
      code="<i class=\\"payment-icon payment-icon-{icon-name}\\" aria-hidden=\\"true\\" /i>"
      type="primary"
    />
  </p>
  <table
    className="table table-plain"
  >
    <thead>
      <tr>
        <th>
          Icon
        </th>
        <th>
          Card type
        </th>
        <th>
          Code
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="payment-card"
      >
        <td>
          <IconPreview
            name="payment-card"
            type="payment-icon"
          />
        </td>
        <td>
          Payment card
        </td>
        <td>
          <CodeTags
            code="payment-card"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="amex"
      >
        <td>
          <IconPreview
            name="amex"
            type="payment-icon"
          />
        </td>
        <td>
          American Expres
        </td>
        <td>
          <CodeTags
            code="amex"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="bank-axept"
      >
        <td>
          <IconPreview
            name="bank-axept"
            type="payment-icon"
          />
        </td>
        <td>
          Bank Accept
        </td>
        <td>
          <CodeTags
            code="bank-axept"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="coop"
      >
        <td>
          <IconPreview
            name="coop"
            type="payment-icon"
          />
        </td>
        <td>
          Coop
        </td>
        <td>
          <CodeTags
            code="coop"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="dankort"
      >
        <td>
          <IconPreview
            name="dankort"
            type="payment-icon"
          />
        </td>
        <td>
          Dankort
        </td>
        <td>
          <CodeTags
            code="dankort"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="diners"
      >
        <td>
          <IconPreview
            name="diners"
            type="payment-icon"
          />
        </td>
        <td>
          Diners Club
        </td>
        <td>
          <CodeTags
            code="diners"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="forbrugsforeningen"
      >
        <td>
          <IconPreview
            name="forbrugsforeningen"
            type="payment-icon"
          />
        </td>
        <td>
          Forbrugsforeningen
        </td>
        <td>
          <CodeTags
            code="forbrugsforeningen"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="ica"
      >
        <td>
          <IconPreview
            name="ica"
            type="payment-icon"
          />
        </td>
        <td>
          ICA
        </td>
        <td>
          <CodeTags
            code="ica"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="ica-banken"
      >
        <td>
          <IconPreview
            name="ica-banken"
            type="payment-icon"
          />
        </td>
        <td>
          ICA Banken
        </td>
        <td>
          <CodeTags
            code="ica-banken"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="ikano-bank"
      >
        <td>
          <IconPreview
            name="ikano-bank"
            type="payment-icon"
          />
        </td>
        <td>
          Ikano Bank
        </td>
        <td>
          <CodeTags
            code="ikano-bank"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="jcb"
      >
        <td>
          <IconPreview
            name="jcb"
            type="payment-icon"
          />
        </td>
        <td>
          JCB
        </td>
        <td>
          <CodeTags
            code="jcb"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="lindex"
      >
        <td>
          <IconPreview
            name="lindex"
            type="payment-icon"
          />
        </td>
        <td>
          Lindex
        </td>
        <td>
          <CodeTags
            code="lindex"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="maestro"
      >
        <td>
          <IconPreview
            name="maestro"
            type="payment-icon"
          />
        </td>
        <td>
          Maestro
        </td>
        <td>
          <CodeTags
            code="maestro"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="mastercard"
      >
        <td>
          <IconPreview
            name="mastercard"
            type="payment-icon"
          />
        </td>
        <td>
          Mastercard
        </td>
        <td>
          <CodeTags
            code="mastercard"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="visa"
      >
        <td>
          <IconPreview
            name="visa"
            type="payment-icon"
          />
        </td>
        <td>
          Visa
        </td>
        <td>
          <CodeTags
            code="visa"
            type="secondary"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <ComponentPreview
    codeFigure={true}
    language="html"
  >
    <i
      aria-hidden="true"
      className="payment-icon payment-icon-payment-card"
    />
    

    <i
      aria-hidden="true"
      className="payment-icon payment-icon-amex"
    />
    

    <i
      aria-hidden="true"
      className="payment-icon payment-icon-bank-axept"
    />
    

    <i
      aria-hidden="true"
      className="payment-icon payment-icon-coop"
    />
    

    <i
      aria-hidden="true"
      className="payment-icon payment-icon-dankort"
    />
    

    <i
      aria-hidden="true"
      className="payment-icon payment-icon-diners"
    />
    

    <i
      aria-hidden="true"
      className="payment-icon payment-icon-forbrugsforeningen"
    />
    

    <i
      aria-hidden="true"
      className="payment-icon payment-icon-ica"
    />
    

    <i
      aria-hidden="true"
      className="payment-icon payment-icon-ica-banken"
    />
    

    <i
      aria-hidden="true"
      className="payment-icon payment-icon-ikano-bank"
    />
    

    <i
      aria-hidden="true"
      className="payment-icon payment-icon-jcb"
    />
    

    <i
      aria-hidden="true"
      className="payment-icon payment-icon-lindex"
    />
    

    <i
      aria-hidden="true"
      className="payment-icon payment-icon-maestro"
    />
    

    <i
      aria-hidden="true"
      className="payment-icon payment-icon-mastercard"
    />
    

    <i
      aria-hidden="true"
      className="payment-icon payment-icon-visa"
    />
    

  </ComponentPreview>
</section>
`;

exports[`Core: Iconography Flags renders 1`] = `
<section>
  <h2
    id="flags"
  >
    Flags
  </h2>
  <p>
    Below is a list of flag icons that we use in our system, To use the flag icons add the classes 
    <CodeTags
      code=".flag-icon"
      type="secondary"
    />
     and  
    <CodeTags
      code=".flag-icon-xx"
      type="secondary"
    />
      (where xx is the ISO_3166-1-alpha-2-code of a country) to an empty 
    <CodeTags
      code="<i>"
      type="secondary"
    />
  </p>
  <table
    className="table table-plain"
  >
    <thead>
      <tr>
        <th>
          Icon
        </th>
        <th>
          Country
        </th>
        <th>
          Code
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="no"
      >
        <td>
          <IconPreview
            name="no"
            size="medium"
            type="flag-icon"
          />
        </td>
        <td>
          Norway
        </td>
        <td>
          <CodeTags
            code="no"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="se"
      >
        <td>
          <IconPreview
            name="se"
            size="medium"
            type="flag-icon"
          />
        </td>
        <td>
          Sweden
        </td>
        <td>
          <CodeTags
            code="se"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="dk"
      >
        <td>
          <IconPreview
            name="dk"
            size="medium"
            type="flag-icon"
          />
        </td>
        <td>
          Denmark
        </td>
        <td>
          <CodeTags
            code="dk"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="fi"
      >
        <td>
          <IconPreview
            name="fi"
            size="medium"
            type="flag-icon"
          />
        </td>
        <td>
          Finland
        </td>
        <td>
          <CodeTags
            code="fi"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="eu"
      >
        <td>
          <IconPreview
            name="eu"
            size="medium"
            type="flag-icon"
          />
        </td>
        <td>
          EU
        </td>
        <td>
          <CodeTags
            code="eu"
            type="secondary"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <ComponentPreview
    codeFigure={true}
    language="html"
  >
    <i
      aria-hidden="true"
      className="flag-icon flag-icon-no"
    />
    

    <i
      aria-hidden="true"
      className="flag-icon flag-icon-se"
    />
    

    <i
      aria-hidden="true"
      className="flag-icon flag-icon-dk"
    />
    

    <i
      aria-hidden="true"
      className="flag-icon flag-icon-fi"
    />
    

    <i
      aria-hidden="true"
      className="flag-icon flag-icon-eu"
    />
    

  </ComponentPreview>
  <h3>
    Sizes
  </h3>
  <p>
    All size available for the flags are pre-defined and are shown below. Tiny is the default size and for example is used input fields. 
  </p>
  <div
    className="d-flex justify-content-between align-items-end p-4"
  >
    <IconPreview
      name="no"
      previewSize={true}
      size="tiny"
      type="flag-icon"
    />
    

    <IconPreview
      name="se"
      previewSize={true}
      size="small"
      type="flag-icon"
    />
    

    <IconPreview
      name="dk"
      previewSize={true}
      size="medium"
      type="flag-icon"
    />
    

    <IconPreview
      name="fi"
      previewSize={true}
      size="large"
      type="flag-icon"
    />
    

    <IconPreview
      name="eu"
      previewSize={true}
      size="huge"
      type="flag-icon"
    />
    

  </div>
  <ComponentPreview
    codeFigure={true}
    language="html"
  >
    <i
      aria-hidden="true"
      className="flag-icon flag-icon-no flag-icon-tiny"
    />
    

    <i
      aria-hidden="true"
      className="flag-icon flag-icon-se flag-icon-small"
    />
    

    <i
      aria-hidden="true"
      className="flag-icon flag-icon-dk flag-icon-medium"
    />
    

    <i
      aria-hidden="true"
      className="flag-icon flag-icon-fi flag-icon-large"
    />
    

    <i
      aria-hidden="true"
      className="flag-icon flag-icon-eu flag-icon-huge"
    />
    

  </ComponentPreview>
  <h3>
    Squared
  </h3>
  <p>
    All flags can be used in a square format. In the example we see squared flags in the size tiny. All predefined sizes are available in combination with 
    <CodeTags
      code=".flag-icon-squared"
      type="secondary"
    />
    . 
  </p>
  <div
    className="d-flex justify-content-between align-items-end p-4"
  >
    <IconPreview
      name="no"
      previewSize={true}
      size="tiny"
      squaredFlag={true}
      type="flag-icon"
    />
    <IconPreview
      name="se"
      previewSize={true}
      size="small"
      squaredFlag={true}
      type="flag-icon"
    />
    <IconPreview
      name="dk"
      previewSize={true}
      size="medium"
      squaredFlag={true}
      type="flag-icon"
    />
    <IconPreview
      name="fi"
      previewSize={true}
      size="large"
      squaredFlag={true}
      type="flag-icon"
    />
    <IconPreview
      name="eu"
      previewSize={true}
      size="huge"
      squaredFlag={true}
      type="flag-icon"
    />
  </div>
  <ComponentPreview
    codeFigure={true}
    language="html"
  >
    <i
      aria-hidden="true"
      className="flag-icon flag-icon-squared flag-icon-no flag-icon-tiny"
    />
    

    <i
      aria-hidden="true"
      className="flag-icon flag-icon-squared flag-icon-se flag-icon-small"
    />
    

    <i
      aria-hidden="true"
      className="flag-icon flag-icon-squared flag-icon-dk flag-icon-medium"
    />
    

    <i
      aria-hidden="true"
      className="flag-icon flag-icon-squared flag-icon-fi flag-icon-large"
    />
    

    <i
      aria-hidden="true"
      className="flag-icon flag-icon-squared flag-icon-eu flag-icon-huge"
    />
    

  </ComponentPreview>
</section>
`;

exports[`Core: Iconography MaterialIcons renders 1`] = `
<section>
  <h2
    id="material-rounded-icons"
  >
    Material Rounded Icons
  </h2>
  <p
    className="mb-0"
  >
    We primarily use the version called Rounded in the Material icons. To aviod a to heavy look, we strive to look for outlined icons. If an icon isn’t available as outlined or filled we can search for it in the Outlined category aswell. For a full overview of the available icons, please, visit the Material.io. 
  </p>
  <a
    className="icon-link"
    href="https://material.io/resources/icons/?style=round"
    rel="noopener noreferrer"
    target="_blank"
  >
    <i
      aria-hidden="true"
      className="material-icons"
    >
      open_in_new
    </i>
    <span
      className="ml-2"
    >
      Material Icons Overview
    </span>
  </a>
  <h3>
    Example of how to implement icons
  </h3>
  <p>
    To use an icon, provide the following markup: 
    <CodeTags
      code="<i class=\\"material-icons\\" aria-hidden=\\"true\\">{icon_name}</i>"
      type="primary"
    />
  </p>
  <div
    className="d-flex justify-content-between p-4"
  >
    <IconPreview
      key="insert_invitation"
      name="insert_invitation"
      preview={true}
      type="material-icons"
    />
    <IconPreview
      key="cake"
      name="cake"
      preview={true}
      type="material-icons"
    />
    <IconPreview
      key="mood"
      name="mood"
      preview={true}
      type="material-icons"
    />
    <IconPreview
      key="thumb_up_alt"
      name="thumb_up_alt"
      preview={true}
      type="material-icons"
    />
    <IconPreview
      key="whatshot"
      name="whatshot"
      preview={true}
      type="material-icons"
    />
  </div>
  <ComponentPreview
    codeFigure={true}
    language="html"
  >
    <i
      aria-hidden="true"
      className="material-icons"
    >
      insert_invitation
    </i>
    

    <i
      aria-hidden="true"
      className="material-icons"
    >
      cake
    </i>
    

    <i
      aria-hidden="true"
      className="material-icons"
    >
      mood
    </i>
    

    <i
      aria-hidden="true"
      className="material-icons"
    >
      thumb_up_alt
    </i>
    

    <i
      aria-hidden="true"
      className="material-icons"
    >
      whatshot
    </i>
    

  </ComponentPreview>
</section>
`;

exports[`Core: Iconography PaymentIcons renders 1`] = `
<section>
  <h2
    id="payment-icons"
  >
    Payment icons
  </h2>
  <p>
    Below is a list of payment icons that we use in our products. Be mindful when using payment logotypes. Try to make the payment icon feel more secondary in combination with the Swedbank Pay logotype. We always want the Swedbank Pay logotype to be primary.
  </p>
  <h3>
    Example of how to implement payment icons
  </h3>
  <p>
    To use an icon, provide the following markup: 
    <CodeTags
      code="<i class=\\"payment-icon payment-icon-{icon-name}\\" aria-hidden=\\"true\\" /i>"
      type="primary"
    />
  </p>
  <table
    className="table table-plain"
  >
    <thead>
      <tr>
        <th>
          Icon
        </th>
        <th>
          Payment method
        </th>
        <th>
          Code
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="viabill"
      >
        <td>
          <IconPreview
            name="viabill"
            size="large"
            type="payment-icon"
          />
        </td>
        <td>
          Viabill
        </td>
        <td>
          <CodeTags
            code="viabill"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="vipps"
      >
        <td>
          <IconPreview
            name="vipps"
            size="large"
            type="payment-icon"
          />
        </td>
        <td>
          Vipps
        </td>
        <td>
          <CodeTags
            code="vipps"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="swish"
      >
        <td>
          <IconPreview
            name="swish"
            size="large"
            type="payment-icon"
          />
        </td>
        <td>
          Swish
        </td>
        <td>
          <CodeTags
            code="swish"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="mobilepay"
      >
        <td>
          <IconPreview
            name="mobilepay"
            size="large"
            type="payment-icon"
          />
        </td>
        <td>
          MobilePay
        </td>
        <td>
          <CodeTags
            code="mobilepay"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="carpay"
      >
        <td>
          <IconPreview
            name="carpay"
            size="large"
            type="payment-icon"
          />
        </td>
        <td>
          CarPay
        </td>
        <td>
          <CodeTags
            code="carpay"
            type="secondary"
          />
        </td>
      </tr>
      <tr
        key="trustly"
      >
        <td>
          <IconPreview
            name="trustly"
            size="large"
            type="payment-icon"
          />
        </td>
        <td>
          Trustly
        </td>
        <td>
          <CodeTags
            code="trustly"
            type="secondary"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <ComponentPreview
    codeFigure={true}
    language="html"
  >
    <i
      aria-hidden="true"
      className="payment-icon payment-icon-viabill"
    />
    

    <i
      aria-hidden="true"
      className="payment-icon payment-icon-vipps"
    />
    

    <i
      aria-hidden="true"
      className="payment-icon payment-icon-swish"
    />
    

    <i
      aria-hidden="true"
      className="payment-icon payment-icon-mobilepay"
    />
    

    <i
      aria-hidden="true"
      className="payment-icon payment-icon-carpay"
    />
    

    <i
      aria-hidden="true"
      className="payment-icon payment-icon-trustly"
    />
    

  </ComponentPreview>
</section>
`;

exports[`Core: Iconography renders 1`] = `
<DocContainer>
  <p
    className="lead"
  >
    At Swedbank Pay we use Material icons from Material Design, but we also include icons for well-known payment providers, and flags for most nations in the world. All icons are integrated and are available in HTML and CSS.
  </p>
  <MaterialIcons />
  <CardIcons />
  <PaymentIcons />
  <Flags />
</DocContainer>
`;
